src/window/basic_frame.rs

```diff
@@ -387,9 +385,12 @@ impl Frame for BasicFrame {
             // grab the current pool
             let pool = self.pools.pool();
             // resize the pool as appropriate
-            let pxcount = 4 * (2 * (BORDER_SIZE * (width + 2 * BORDER_SIZE))
-                + 2 * (BORDER_SIZE * (height + 2 * BORDER_SIZE))
-                + (HEADER_SIZE * width));
+            let pxcount = if (width + 2 * BORDER_SIZE) > (height + HEADER_SIZE) {
+                (HEADER_SIZE * width) + ((width + 2 * BORDER_SIZE) * BORDER_SIZE)
+            } else {
+                (HEADER_SIZE * width) + ((height + HEADER_SIZE) * BORDER_SIZE)
+            };
+
```

Comment:
This can be factored using `std::cmp::max()`, and would improve readability I believe:

```rust
use std::cmp::max;

let pxcount = HEADER_SIZE * width +
              BORDER_SIZE * max(height + HEADER_SIZE, width + 2 * BORDER_SIZE);
```